docs/data-sources/region.md

```diff
@@ -1,33 +1,49 @@
----
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "biganimal_region Data Source - terraform-provider-biganimal"
-subcategory: ""
-description: |-
-  The available regions within a cloud provider
----
-
 # biganimal_region (Data Source)
-
-The available regions within a cloud provider
-
-
+The region data source shows the available regions within a cloud provider.
+
+## Example Usage
+```terraform
+variable "cloud_provider" {
+  type        = string
+  description = "Cloud Provider"
+
+  validation {
+    condition     = contains(["aws", "azure"], var.cloud_provider)
+    error_message = "Please select one of the supported regions: aws, azure."
+  }
+}
+
+data "biganimal_region" "this" {
+  cloud_provider = var.cloud_provider
+  // region_id   = "us-west-1" //optional
+  // query       = "eu" // optional
+}
+
+output "regions" {
+  value = data.biganimal_region.this.regions
+}
+
+output "cloud_provider_id" {
+  value = data.biganimal_region.this.cloud_provider
+}
+```
 
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Required
 
-- `cloud_provider` (String) Cloud Provider
+- `cloud_provider` (String) Cloud Provider to list the regions. e.g. "aws", "azure".
 
 ### Optional
 
-- `query` (String) Query to filter region list
-- `region_id` (String) Unique Region ID
+- `query` (String) Query to filter region list.
+- `region_id` (String) Unique Region ID. e.g. "germanywestcentral" in the azure cloud provider, "eu-west-1" in the aws cloud provider.
```

Comment:
```suggestion
- `region_id` (String) Unique Region ID. For example, "germanywestcentral" in the Azure cloud provider, "eu-west-1" in the AWS cloud provider.
```

Reply:
The `e.g.` -> `For example` is fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/4de4c956dd0831008781a5f59b9011542e378c5a
The spelling of the cloud providers is fixed at https://github.com/EnterpriseDB/terraform-provider-biganimal/commit/fd43c8ae86d3aad7abff08c0cece786c2e54a182